### PR TITLE
Add osLoc for global profiles in `ProfileInfo` APIs

### LINKED ISSUE #771: [next] ProfileInfo API not returning an osLoc
When calling `ProfileInfo.getAllProfiles()`, the profiles in the global layer didn't have an osLoc specified, which prevented some Applications to be able to load all default profiles on load (e.g. Zowe Explorer)

Source code:
https://github.com/zowe/imperative/blob/e804c8bdd85f7704fcc0001548dc007bc7695b9e/packages/config/src/ProfileInfo.ts#L1190-L1191

By removing `this.mLoadedConfig.mActive.global === layer.global` from the condition, it allows the `findTeamOsLocation` to return all `osLoc`s where the given profile is found.

This may have unintended consequences to application developers since it may cause the application to load all default profiles it can find (including default ones in the global layer)

---

**API Call:**
Before
image

After
image

**Zowe Explorer Implementation:**
Before
image

After
image

### PR DESCRIPTION
**Pull request by zFernand0**
In this PR:
- Fix #771

### CODE REVIEW

**Review by awharn — APPROVED** [review]

File: packages/config/__tests__/__resources__/ProfInfoApp_home_team_config_proj/ProfInfoApp.schema.json [hunk 1]
```diff
@@ -0,0 +1,393 @@
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "$version": 0.4,
```
Comment (t1m0thyj) [inline packages/config/__tests__/__resources__/ProfInfoApp_home_team_config_proj/ProfInfoApp.schema.json:3]: Not sure what order we plan to merge PRs in, but we should update this version to `"1.0"`.

**Review by t1m0thyj — APPROVED** [review]

File: packages/config/__tests__/__resources__/ProfInfoApp_home_team_config_proj/ProfInfoApp.schema.json (hunk 1 above)
Comment (zFernand0) [reply packages/config/__tests__/__resources__/ProfInfoApp_home_team_config_proj/ProfInfoApp.schema.json:3]: if we merge #778 before this one (which I would advise we do), then I don't mind updating it here 😋

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,12 @@
 
 All notable changes to the Imperative package will be documented in this file.
 
+## Recent Changes
+
+- BugFix: Fixed missing `osLoc` information from `ProfileInfo.getAllProfiles()`. [#771](https://github.com/zowe/imperative/issues/771)
+- BugFix: Fixed updateKnownProperty svaing to the active layer instead of the layer of the desired profile.
```
Comment (t1m0thyj) [inline CHANGELOG.md:8]: `svaing` -> `saving`

File: packages/config/src/doc/IProfInfoProps.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+/*
```
Comment (t1m0thyj) [inline packages/config/src/doc/IProfInfoProps.ts:1]: Please fix lint errors in this file 🙂

**Review by t1m0thyj — APPROVED** [review]

**Pull request merged**